vertical-pod-autoscaler/enhancements/7862-cpu-startup-boost/README.md

```diff
@@ -0,0 +1,283 @@
+   * The VPA autoscaling mode must be `InPlaceOrRecreate` (since it does not
+   make sense to use this feature with disruptive modes of VPA).
+   * The boost factor is >= 1 (via CRD validation rules)
+   * Only one of `StartupBoost.CPU.Factor` or `StartupBoost.CPU.Value` is
+   specified
+   * The [feature enablement](#feature-enablement) flags must be on.
+
+
+#### Dynamic Validation
+
+* `StartupBoost.CPU.Value` must be greater than the CPU request or limit of the
+  container during the boost phase, otherwise we risk downscaling the container.
+
+* Workloads must be configured with a Readiness or a Startup probe to be able to
+utilize this feature. Therefore, VPA will not boost CPU resources of workloads
+that do not configure a Readiness or a Startup probe.
+
+### Mitigating Failed In-Place Downsizes
+
+The VPA Updater **will not** evict a pod to actuate a startup CPU boost
+recommendation if it attempted to apply the recommendation in place and it
+failed (see the [scenarios](https://github.com/kubernetes/autoscaler/blob/0a34bf5d3a71b486bdaa440f1af7f8d50dc8e391/vertical-pod-autoscaler/enhancements/4016-in-place-updates-support/README.md?plain=1#L164-L169 ) where the VPA
+updater will consider that the update failed). This is to avoid an eviction
+loop:
+
+1. A pod is created and has its CPU resources boosted
+1. The pod is ready. VPA Updater tries to downscale the pod in-place and it
+fails.
+1. VPA Updater evicts the pod. Logic flow goes back to (1).
+
+### Feature Enablement and Rollback
+
+#### How can this feature be enabled / disabled in a live cluster?
+
+* Feature gates names: `CPUStartupBoost` and `InPlaceOrRecreate` (from
+[AEP-4016](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/enhancements/4016-in-place-updates-support/README.md#feature-enablement-and-rollback))
+* Components depending on the feature gates:
+  * admission-controller
+  * updater
+
+Enabling of feature gates `CPUStartupBoost` AND `InPlaceOrRecreate` will cause
+the following to happen:
+  * admission-controller to **accept** new VPA objects being created with
+`StartupBoostOnly` configured.
+  * admission-controller to **boost** CPU resources.
+  * updater to **unboost** the CPU resources.
+
+Disabling of feature gates `CPUStartupBoost` OR `InPlaceOrRecreate` will cause
+the following to happen:
+  * admission-controller to **reject** new VPA objects being created with
+  `StartupBoostOnly` configured.
+    * A descriptive error message should be returned to the user letting them
+    know that they are using a feature gated feature.
+  * admission-controller **to not** boost CPU resources, should it encounter a
+  VPA configured with a `StartupBoost` config and `StartupBoostOnly` or `Auto`
+  `ContainerScalingMode`.
+  * updater **to not** unboost CPU resources when pods become `Ready`, should it
+  encounter a VPA configured with a `StartupBoost` config and `StartupBoostOnly`
+  or `Auto` `ContainerScalingMode`.
```

Comment:
I assume `Auto` should actually be `InPlaceOrRecreate` here? Since at the moment, we're not touching the Auto mode

Reply:
Wrong Auto, ignore me!